### ISSUE TITLE
Google Search shows my real location based on my IP while using Xray

### ISSUE THREAD

**Issue by Evv1L** [XTLS/Xray-core#1279]
## TL;DR
Google Search shows my real location while using Xray + Vless + XTLS + Nginx server on Android and Linux, but other websites and IP tests shows my VPS IP where Xray server is located. Is this hardcoded behavior?

## Expected Behavior
I expect Google Search (and other Google product pages) to say that I'm (for example) in Netherlands while using Xray.

## Current Behavior
Google Search shows the city and country I am in "From your IP address" at the bottom of search results page while using Xray. **Sometimes Google says correct city, but completely wrong city district.**

All other websites (and search engines) says that I'm in Netherlands. DNS Leak tests shows DNS servers only from Netherlands and IP Leak tests shows only Netherlands IP where my VPS is located.

## Steps to Reproduce
I'm using VPS with Debain 11:
1. Enable BBR
2. Install and configure Nginx:
```nginx
server {
	listen 80 default_server;
	listen [::]:80 default_server;

	root /var/www/html;
	index index.html index.htm index.nginx-debian.html;

	server_name DOMAIN;

	location / {
		try_files $uri $uri/ =404;
	}
}
```

3. Get _only_ SSL certificate using Certbot 
```bash
certbot certonly --nginx
```

4. Install Xray 1.6.0 server:
```bash
bash -c "$(curl -L https://github.com/XTLS/Xray-install/raw/main/install-release.sh)" @ install -u root
```

* Xray server config:
```json
{
    "log": {
        "loglevel": "warning"
    },
    "inbounds": [
        {
            "port": 443,
            "protocol": "vless",
            "settings": {
                "clients": [
                    {
                        "id": "##################################",
                        "flow": "xtls-rprx-direct",
                        "level": 0,
                        "email": "love@example.com"
                    }
                ],
                "decryption": "none",
                "fallbacks": [
                    {
                        "dest": 80
                    }
                ]
            },
            "streamSettings": {
                "network": "tcp",
                "security": "xtls",
                "xtlsSettings": {
                    "alpn": [
                        "http/1.1"
                    ],
                    "certificates": [
                        {
                            "certificateFile": "/etc/letsencrypt/live/DOMAIN/fullchain.pem",
                            "keyFile": "/etc/letsencrypt/live/DOMAIN/privkey.pem"
                        }
                    ]
                }
            }
        }
    ],
    "outbounds": [
        {
            "protocol": "freedom"
        }
    ]
}

```

5. Configure Xray Linux client
* Create Xray config
```json
{
    "log": {
        "loglevel": "warning"
    },
    "inbounds": [
        {
            "port": 10800,
            "listen": "127.0.0.1",
            "protocol": "socks",
            "settings": {
                "udp": true
            }
        }
    ],
    "outbounds": [
        {
            "protocol": "vless",
            "settings": {
                "vnext": [
                    {
                        "address": "VPS_IP_ADDRESS",
                        "port": 443,
                        "users": [
                            {
                                "id": "###########################################",
                                "flow": "xtls-rprx-direct",
                                "encryption": "none",
                                "level": 0
                            }
                        ]
                    }
                ]
            },
            "streamSettings": {
                "network": "tcp",
                "security": "xtls",
                "xtlsSettings": {
                    "serverName": "DOMAIN_NAME"
                }
            }
        }
    ]
}

```

* Run Xray from config:
```bash
./xray -config=config.json
```

* Configure Firefox to use Xray: Enable SOCKS5 proxy with address 127.0.0.1:10800 and Enable DNS through SOCKS5 proxy.

7. Configure Android client:
- Install v2rayNG
- Create VLESS config with following parameters:
```python
Remarks: Any name
Address: VPS IP address
Port: 443
ID: Your UUID
Flow: xtls-rprx-direct
Encryption: none

Network: tcp
Head type: none
Request host И Path - empty
TLS: xtls
SNI: DOMAIN.NAME
allowInsecure: false
```

* (**OR** import custom config from Linux client)
* Run v2rayNG in VPN mode

8. Visit [Google Search](https://google.com) and IP/DNS leak test sites like [Whoer](https://whoer.net) or [IPLeak](https://ipleak.net). Google should say your location "From your IP address", but all other sites should show you your Xray server address.

## Other notes
* I don't live in China.
* I expect to route ALL (except localhost) my traffic to my Xray server
* Linux client uses Cloudflare DNS in `/etc/resolv.conf`
* I tried to use Shadowsocks + Cloak on the same VPS and everything worked fine - all sites shows the same VPS IP as expected.
* I'm using Firefox with uBlock Origin on both Linux and Android clients.
* I'm visiting Google Search in Incognito mode and I don't have any Google accounts on Android device.
* Device location and location permissions in Firefox are also disabled.

**Comment by SakuraSakuraSakuraChan** [comment]
It has nothing to do with xray or any proxy affairs. Google has it own way to update GeoIP database by tracking user's location based on phone's location permission, and it can be affected by neighboring IP address

https://vps.dance/google-ip-location.html
try to read by google translate

**Comment by MHSanaei** [comment]
I think if you turn on the proxy in Cloudflare DNS, maybe cloud work for you, but I already tested that yesterday, and unfortunately, after I turned it on, vmess didn't connect anymore, so I back it to normal

**Comment by Evv1L** [comment]
> It has nothing to do with xray or any proxy affairs. Google has it own way to update GeoIP database by tracking user's location based on phone's location permission, and it can be affected by neighboring IP address
> 
> https://vps.dance/google-ip-location.html try to read by google translate

Yea, I configured OpenVPN on the same VPS and issue is still here. It's not Xray issue. 
Thank you for response!
